source/common/http/filter_manager.h

```diff
@@ -812,16 +815,17 @@ class FilterManager : public ScopeTrackedObject,
 
   struct State {
     State()
-        : remote_complete_(false), local_complete_(false), has_continue_headers_(false),
-          created_filter_chain_(false), is_head_request_(false), is_grpc_request_(false),
-          non_100_response_headers_encoded_(false) {}
+        : remote_complete_(false), local_complete_(false), encoding_complete_(false),
+          has_continue_headers_(false), created_filter_chain_(false), is_head_request_(false),
+          is_grpc_request_(false), non_100_response_headers_encoded_(false) {}
 
     uint32_t filter_call_state_{0};
 
     bool remote_complete_ : 1;
     bool local_complete_ : 1; // This indicates that local is complete prior to filter processing.
-                              // A filter can still stop the stream from being complete as seen
-                              // by the codec.
+    // A filter can still stop the stream from being complete as seen
+    // by the codec.
+    bool encoding_complete_ : 1;
```

Comment:
this comment made me think that encoding complete would go from true to false, if some filter retroactively made the stream not complete, where I *think* what actually happens is sometimes we set local_complete_ true before we set encoding_complete_

Maybe we can update the comments a bit to explain the difference more clearly?